src/sas/sasgui/perspectives/fitting/basepage.py

```diff
@@ -1686,7 +1692,12 @@ def _reset_strparam_state(self, listtorestore, statelist):
                 value = item_page_info[2]
                 selection = value
                 if value in self.model.fun_list:
-                    selection = self.model.fun_list[value]
+                    # to fix: fun_list is not a dictionary, so 
+                    # I commented the following original line 
+                    #selection = self.model.fun_list[value]
+                    # and replaced by this.
+                    # I think this should work, but could not test it.
+                    selection = self.model.fun_list.index(value)
```

Comment:
_reset_strparam_state is never referenced.  Maybe put an exception or a logging statement at the beginning while you are testing to see if it is being triggered in some other way: 
```
    logger.error("calling _reset_strparam_state")
    raise Exception("unused function")
```